Sprint-3/4-stretch-investigate/password-validator.js

```diff
@@ -1,5 +1,26 @@
 function passwordValidator(password) {
-    return password.length < 5 ? false : true
+    const passwords = ["Test&1", "P@ssw0rd!", "$ample5"];
+
+    if (password.length < 5) return false;
+
+    let upperCase = false;
+    let lowerCase = false;
+    let numberChar = false;
+    const specialChar = "!£%$&*?#";
+    let special = false;
+
+    for (let char of password){
+        if (char >= "A" && char <="Z") upperCase = true;
+        else if (char >="a" && char <="z") lowerCase = true;
+        else if (char >= "0" && char <= "9") numberChar = true;
+        else if (specialChar.indexOf(char) !== -1) special = true;
+    }
```

Comment:
this is a really good way to check  characters. Also It is an ideal way to define your functions.  but you can  define your variables inside of your function in front of your if check.